src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs

```diff
@@ -67,9 +70,67 @@ bool interceptor(ChainedHeaderBlock chainedHeaderBlock)
 
                 // This will cause the reorg chain to fail at block 8 and roll back any changes.
                 TestHelper.WaitLoop(() => TestHelper.IsNodeSyncedAtHeight(minerA, 9));
-                TestHelper.WaitLoop(() => TestHelper.IsNodeSyncedAtHeight(minerB, 10));
                 TestHelper.WaitLoop(() => TestHelper.IsNodeSyncedAtHeight(minerC, 9));
             }
         }
+
+        /// <summary>
+        /// In this scenario we test what happens when a node disconnected during a rewind and before it
+        /// tries to connect to another chain with longer chain work.
+        /// </summary>
+        [Fact]
+        public void ReorgChain_AfterInitialRewind_ChainB_MinerB_Disconnects()
+        {
+            using (var builder = NodeBuilder.Create(this))
+            {
+                var minerA = builder.CreateStratisPosNode(new StratisRegTest()).WithDummyWallet();
+                var minerB = builder.CreateStratisPosNode(new StratisRegTest()).WithDummyWallet();
+                var minerC = builder.CreateStratisPosNode(new StratisRegTest()).WithDummyWallet();
+
+                // Configure the interceptor to disconnect a node after a certain block has been disconnected (rewound).
+                bool interceptor(ChainedHeaderBlock chainedHeaderBlock)
+                {
+                    if (chainedHeaderBlock.ChainedHeader.Previous.Height == 5)
+                    {
+                        // Ensure that minerA's tips has rewound to 5.
+                        TestHelper.WaitLoop(() => TestHelper.IsNodeSyncedAtHeight(minerA, 5));
+                        TestHelper.Disconnect(minerA, minerC);
+                        return true;
+                    }
+
+                    return false;
+                }
+
+                minerA.BlockDisconnectInterceptor(interceptor);
+
+                // Start the nodes.
+                minerA.Start();
+                minerB.Start();
```

Comment:
miner B never mines perhaps better to keep the terminology of MinerA, MinerB and Syncer

Reply:
Ah yes, good spot. Will rename 👍